.github/workflows/pipeline.yml

```diff
@@ -42,6 +42,30 @@ jobs:
               run: pnpm run build
             - name: Run unit tests
               run: pnpm run test
+            - name: Check for changes
+              uses: dorny/paths-filter@v2
+              id: check_changes
+              with:
+                filters: |
+                  src:
+                    - 'packages/preview-middleware/**'
+              # To include changes for other packages, simply add this  - 'packages/<package-name>/**'
+            - name: Install playwright browsers
+              if: always() && steps.check_changes.outputs.src == 'true'
+              run: npx playwright install --with-deps
+            - name: Run integration tests
+              # Run only if preview-middleware was changed
+              if: always() && steps.check_changes.outputs.src == 'true'
+              run: pnpm run test:integration
+            - name: Upload playwright reports
+              uses: actions/upload-artifact@v3
+              if: always() && steps.check_changes.outputs.src == 'true'
+              with:
+                name: playwright-reports-os-${{ matrix.os }}-node-version-${{ matrix.node-version }}
+                path: |
+                  packages/*/playwright-report
+                  !packages/*/node_modules
+                retention-days: 15
```

Comment:
@mr25mr Have you tried to use NX caching instead of detecting changes?
See test for example:
https://github.com/SAP/open-ux-tools/blob/main/nx.json#L9
https://github.com/SAP/open-ux-tools/blob/main/nx.json#L42-L52

Also with node_module caching we can probably skip `npx playwright install --with-deps`
https://github.com/SAP/open-ux-tools/blob/main/.github/workflows/pipeline.yml#L26-L34

Reply:
@devinea I tried to use NX caching, does not seems to be straightforward to achieve same goal. We need to have playwright reports for each os with combination of node version e.g `playwright-reports-os-windows-latest-node-version-18.x` and using NX caching for this goal seemed a bit complicated.

regarding `npx playwright install --with-deps` we need to install browser dependencies due to this issue https://github.com/SAP/open-ux-tools/pull/1507/files#diff-0fc4e4b47236265ffb9c3ec977c54445ee6e60856bd910f44232abe7a91e1316R41
Here is official doc. https://playwright.dev/docs/ci-intro#setting-up-github-actions

one other constrains - [Each version of Playwright needs specific versions of browser binaries to operate. You will need to use the Playwright CLI to install these browsers](https://playwright.dev/docs/browsers#introduction). Caching them, not sure.

Reply:
As discussed https://github.com/SAP/open-ux-tools/pull/1517/commits/e3d0dcdae7a00e1af47a3906b9359496b7f3c66c seems like a viable approach using NX.
Playwright installed is not cached with that approach. But something like this looks nice https://github.com/microsoft/playwright/issues/7249#issuecomment-1385567519

Reply:
@devinea address in this commit https://github.com/SAP/open-ux-tools/pull/1507/commits/c889140a1b5509489577f01165fe1cf09e93dae3
Currently playwright is installed in mono repo only once, and when building [key for caching  ](https://github.com/SAP/open-ux-tools/pull/1507/files#diff-fe700949ba86403864aa7b6fdcab2c00c55e1047f73a6a282135a35e8a21e03bR54) we do not consider playwright version. It should be ok as long as we do not use different playwright versions. Getting playwright version to include it in key, does not work reliable on windows at least I tried a couple of time :)